tests/ws_integration.rs

```diff
@@ -235,6 +236,106 @@ mod ws_integration {
 		Ok(())
 	}
 
+	#[test(tokio::test)]
+	async fn variable_auth_live_query() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
I've the feeling this test may hide bugs at some point in the future. It assumes that if a message is not received after 2 seconds, it's because the auth check prevented the notification from arriving to the client, but in reality the reason could be something else. That "something else" could hide an auth bug.

Don't want to delay this PR more, but given it's security related, maybe it's worth taking a bit more time to get this test right: you may come up with a more reliable way, but I would probably create two LQ targeting the same table, one of them with the session expiring in 2s and the other one with the session expiring in 10s, and then I would wait those 10s for both, and verify I received it for 1 and not for the other.

This new implementation could still be affected by the situation I mentioned at the beginning, but given you have a second LQ with the same characteristics running in parallel inside the same test, it's more unlikely that one of the queries is delayed so much more than the other one for external reasons.

Just my 2cts

Reply:
Yes, good idea - created SUR-262